lib/fndoc.nix

```diff
@@ -0,0 +1,15 @@
+{
+  docFn = {
+    description,
+    fn,
+    examples ? [],
+    example ? null
+  }:
+  let documentedFunction =
+    {
+      type = "documentedFunction";
+      __functor = self: x: fn x;
```

Comment:
@zimbatm I was also curious about `__functor` performance so I made a quick little benchmark [here](https://gist.github.com/ericsagnes/d3770bc5e04328f943d9e60d4e64e0d4). (Not sure if it is the right way to benchmark nix function calls and if the results have any relevance)

Anyway, the results show that the use of functor increased the number of cycles by about 3%.